modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/JSONFrontendDatasetEntries.macro

```diff
@@ -36,6 +36,48 @@ definition {
 		}
 	}
 
+	macro _createFDSViewViaAPI {
+		Variables.assertDefined(parameterList = "${datasetName},${dataSetViewName}");
+
+		var portalURL = JSONCompany.getPortalURL();
+
+		if (!(isSet(userEmailAddress))) {
+			var userEmailAddress = "test@liferay.com";
+		}
+
+		if (!(isSet(userPassword))) {
+			var userPassword = "test";
+		}
+
+		if (!(isSet(dataSetViewDescription))) {
+			var dataSetViewDescription = "";
+		}
+
+		var datasetEntryId = JSONFrontendDatasetEntries._getDatasetEntryIdByName(datasetName = ${datasetName});
```

Comment:
got it. Awesome!

Reply:
Yes I thought of that but in this case we would have to add 2 loops in series, because we would need 1 for the datasetID and another for the view names. To create a view it is good to specify in the test the input it is related to directly related to an entry.